src/raft/raft.go

```diff
@@ -141,21 +141,23 @@ func (rf *Raft) RequestVote(reqVoteArgs *RequestVoteArgs, reqVoteReply *RequestV
 	defer rf.mtx.Unlock()
 
 	if rf.currentTerm > reqVoteArgs.Term {
+		log.Debug("Raft.RequestVote", "rf.currentTerm(%d) > reqVoteArgs.Term(%d)", rf.currentTerm, reqVoteArgs.Term)
 		reqVoteReply.Term = rf.currentTerm
 		return
 	}
-	if rf.moreUpToDate(reqVoteArgs.LastLogTerm, reqVoteArgs.LastLogIndex) {
-		// log.Debug("Raft.RequestVote", "lastLog={%+v}, args.LastLogTerm=%d, args.LastLogIndex=%d", *rf.getLastLogEntry(), reqVoteArgs.LastLogTerm, reqVoteArgs.LastLogIndex)
-		return
-	}
 
-	reqVoteReply.Term = reqVoteArgs.Term
 	if rf.currentTerm < reqVoteArgs.Term || rf.votedFor == NullCandidate || rf.votedFor == reqVoteArgs.CandidateId {
 		log.Debug("Raft.RequestVote", "raft server %d voted for %d previously, votes for the candidate %d", rf.myIdx, rf.votedFor, reqVoteArgs.CandidateId)
+		rf.demote(reqVoteArgs.Term)
+
+		if rf.moreUpToDate(reqVoteArgs.LastLogTerm, reqVoteArgs.LastLogIndex) {
```

Comment:
Updated this `RequestVote` to pass `TestBackup2B` finally, by moving this `moreUpToDate` function:  
The old logic ignored to handle the edge case where the follower has a more up-to-date logs while it's `currentTerm` is less than the candidate's, which means it's `currentTerm` couldn't get updated promptly.